### DISCUSSION TITLE
Q&A on setting up the project locally

### DISCUSSION THREAD

**Discussion by sadath2001** [AlfiyaSiddique/TastyTrails#37] [Q&A]
Hi, This is a space to ask questions on setting up the project locally

**Comment by krishashah-03** [comment]
Hi, @sadath2001 @AlfiyaSiddique , after cloning the repo the backend got set perfectly  but in frontend only main-page is loafing ,if I'm clicking any other link the blank page is coming as I'm showing in the video.
After inspecting it showed this line: Access to XMLHttpRequest at 'http://localhost:8080/api/recipes' from origin 'http://localhost:5173' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
I'm here attaching the video also :

https://github.com/user-attachments/assets/0f7b7f00-f474-48ea-becb-f7fd92408f8a

https://github.com/user-attachments/assets/a352995f-0e4b-4fc2-a44e-94c0201cea78

I would like you to look into this matter as without solving this issue I might not be able to work on adding new feature!

**Comment by sadath2001** [comment]
Hey @krishashah-03 ,, you should add localhost:5173 in server.js file as well. Did you do that?

**Reply by krishashah-03** [reply]
Hii @sadath2001 Yes, I actually didn't set the port in server.js but even after seeting the port now, I'm still not getting the images 
Now this errors are coming:
![image](https://github.com/user-attachments/assets/8caba5fd-c387-458b-b405-5717c3ee65c4)
![image](https://github.com/user-attachments/assets/7892a3ba-1655-493e-bc92-9108fb910e12)
Can you tell me how to solve this react errors? 
Bte thankyou for helping me solving this one problem!

**Reply by sadath2001** [reply]
To solve this you have to add Google client Id in the front end .env file.

Fire up Google to check how to generate a Google client Id(for Google o auth login)
And paste it there 

In back end make sure to add your GitHub credentials such as repo(for recipe image pushing), branch and also your GitHub access token(can ge generated via your GitHub settings)

This should solve your error.

**Comment by sadath2001** [comment]
server.js:
![image](https://github.com/user-attachments/assets/a0092fa4-6a6b-4afa-adb1-b8463e18c3c9)

**Comment by sushil-sagar05** [comment]
Hey @sadath2001 i put all the answers to .env in my backend and in fronted i also did client id part but nothing happens.. Please help me with this

**Reply by sadath2001** [reply]
@sushil-sagar05 the screenshot of the error that you're getting

**Reply by sushil-sagar05** [reply]
![gssoc-issue-5](https://github.com/user-attachments/assets/bea98103-d42b-42a6-a4ad-e8c7a701473f)
@sadath2001

**Reply by sadath2001** [reply]
@sushil-sagar05 the first error. Can you send it's screenshot?
Youve sent half

**Reply by sushil-sagar05** [reply]
![gssoc-6](https://github.com/user-attachments/assets/5370f028-37b6-4ac8-8cbe-75a31fff4583)
@sadath2001

**Reply by sadath2001** [reply]
@sushil-sagar05 , not sure on this one, I'll get back to you by tomorrow.

**Comment by sushil-sagar05** [comment]
any update @sadath2001 ? please solve my problem

**Reply by sadath2001** [reply]
@sushil-sagar05 ,i feel this is more from a server side error.
if you are using mongodb cloud, make sure that your ip is whitelisted, you can ignore the first error as of now.
rest i think everything is good.

**Reply by sushil-sagar05** [reply]
hey @sadath2001 i did whitelisting my current ip but same axios 500 error coming and in terminal this error showing
![gssoc-issue-6](https://github.com/user-attachments/assets/f6d3ed45-b9ae-4847-9c9b-3d7140db50ee)

**Reply by sadath2001** [reply]
you have added 5174 as port in allowed origins (server.js line 20), are you sure that your front end is running on that port?

**Reply by sushil-sagar05** [reply]
> you have added 5174 as port in allowed origins (server.js line 20), are you sure that your front end is running on that port?

yesss

**Reply by sadath2001** [reply]
@sushil-sagar05 ,
can u send me the screenshot of your front end console.
also, it should be http://localhost:5174, without a /

**Reply by sadath2001** [reply]
> @sushil-sagar05 , can u send me the screenshot of your front end console. also, it should be http://localhost:5174, without a /

did you do this?

**Reply by sushil-sagar05** [reply]
yess man i did , basically i want to add slider properties to world best dishes section in landing page but i am not able to connect my database here and donot access all the recipes list :<

**Comment by sushil-sagar05** [comment]
![gssoc-issue-7](https://github.com/user-attachments/assets/23f5926f-e328-453b-882d-21a2cc0d342f)

@sadath2001

**Reply by sadath2001** [reply]
> ![gssoc-issue-7](https://private-user-images.githubusercontent.com/169032900/378116161-23f5926f-e328-453b-882d-21a2cc0d342f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkzNjczMTEsIm5iZiI6MTcyOTM2NzAxMSwicGF0aCI6Ii8xNjkwMzI5MDAvMzc4MTE2MTYxLTIzZjU5MjZmLWUzMjgtNDUzYi04ODJkLTIxYTJjYzBkMzQyZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxOVQxOTQzMzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZGUwY2NiOGVkMzBhMmE0ZjVhNTViN2RmNGNiYTdhMzc0MWEyMTljYmU5ZGQxNTk2ZTFiZWE5Y2EzOTNhODgwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.PY0eLxR6WWhHblg2uFAhXYj4KJJyNVaYyD3fp1twUKY)
> 
> @sadath2001

@sushil-sagar05 , then even im not sure about this. the solutions that i suggested were all i could think of.

**Comment by AlfiyaSiddique** [comment]
@sushil-sagar05 
Please check the following things one.
1. The MongoDB URL is in .env, so console.log and check if the URL is getting retrieved from .env file to others files correctly or not.
2. If the point 1 is working fine go to your monogdb atlas and check that the network access have 0.0.0.0 IP address is added, a reference is given below:
![image](https://github.com/user-attachments/assets/7d143a9d-a1bb-4c39-98f9-b6be6fe01de8)


3. If both of the above points is correct, check the MongoDB URL, it should be in this format:
`mongodb+srv://<username>:<password>@<cluster>.<random characters>.mongodb.net/<database_name>?retryWrites=true&w=majority`

**Reply by sushil-sagar05** [reply]
Hey @AlfiyaSiddique  yesss the 3rd point is the problem i don't write  database name in mongodburi, But the problem is after that i am not able to get the world's best dishes section 
![TastyTrails-issue](https://github.com/user-attachments/assets/80e000fc-3549-4087-8ba7-41ab9d2289eb)

**Comment by AlfiyaSiddique** [comment]
@sushil-sagar05 That is because you do not have any recipes in your database. In deployed link, the app is connected to the actual database but in local you are connecting to your own database. So add some recipes in the local app, and you will get the world's best dishes section. Let me know if this works
